packages/core/src/services/providerApiKeys/create.ts

```diff
@@ -0,0 +1,36 @@
+import { Providers } from '$core/browser'
+import { database } from '$core/client'
+import { Result, Transaction } from '$core/lib'
+import { providerApiKeys, Workspace } from '$core/schema'
+
+export function createProviderApiKey(
+  {
+    workspace,
+    provider,
+    token,
+    name,
+    authorId,
+  }: {
+    workspace: Partial<Workspace>
+    provider: Providers
+    token: string
+    name: string
+    authorId: string
+  },
+  db = database,
+) {
+  return Transaction.call(async (tx) => {
+    const result = await tx
+      .insert(providerApiKeys)
+      .values({
+        workspaceId: workspace.id!,
+        provider,
+        token,
+        name,
+        authorId,
+      })
+      .returning()
```

Comment:
I know we want to return anything here. However, we must be mindful because this can leak to the frontend columns, which might not be necessary or desirable for the client. We solved this before with presenters or POJOs. Not sure if it's worth it just commenting

Reply:
Yup,

any serialization concerns should happen at the boundaries (actions/controllers)